### PR TITLE
test: allow to build a custom docker image

### PR DESCRIPTION
**Pull request by philpep**
Invocation example:

```
$ tree demo
demo
├── entrypoint.sh
├── pillar
│   ├── saemref.sls
│   └── top.sls
└── salt
    └── top.sls

$ cat demo/entrypoint.sh
set -e

/etc/init.d/cron start
/etc/init.d/redis-server start
supervisord --nodaemon

$ cat demo/salt/top.sls
base:
  '*':
    - saemref
    - saemref.supervisor

$ ./run-tests.py build jessie --tag saemref-demo --file-root demo/salt \
	--pillar-root demo/pillar --salt \
        --docker-add-cmd 'ADD demo/entrypoint.sh /entry point.sh\nCMD sh /entrypoint.sh\nEXPOSE 8080\n'

$ docker run -d -P saemref-demo
```

### CODE REVIEW

**Comment by philpep** [comment]
Added an option to prepend docker instruction too

File: run-tests.py [hunk 1]
```diff
@@ -41,25 +43,44 @@ def cli():
...
-        "docker", "build", "-t", tag, "-f", dockerfile, ".",
-    ])
+@click.option('--tag', default=None)
+@click.option('--pillar-root', type=click.Path(exists=True), default='test/pillar')
+@click.option('--file-root', type=click.Path(exists=True), default='test/salt')
+@click.option('--docker-add-cmd', type=click.UNPROCESSED, default=None, help=(
+    "Custom docker instruction to add when building the image (eg. EXPOSE 8080)"
+))
+def build(image, salt, tag, pillar_root, file_root, docker_add_cmd):
+    dockerfile = tempfile.NamedTemporaryFile(
```
Comment (dlax) [inline run-tests.py]: `NamedTemporaryFile` can be used as a context manager, which handles deletion upon exit

**Comment by dlax** [comment]
not sure we really want the pseudo-templating of Dockerfile. just let the user provide its own base dockerfile and extend it in `build` command. passing `pillar-root` and `file-root` seems like a good idea though

**Comment by philpep** [comment]
Agree. Removed docker append/prepend options.

**Pull request merged**